crates/fuel-core/src/graphql_api/worker_service.rs

```diff
@@ -381,6 +386,64 @@ where
             )
             .into());
         }
+
+        for receipt in result.receipts() {
+            match receipt {
+                Receipt::Mint {
+                    sub_id,
+                    contract_id,
+                    val,
+                    ..
+                } => {
+                    let asset_id = contract_id.asset_id(sub_id);
+                    let current_count = match db.storage::<AssetsInfo>().get(&asset_id) {
+                        Ok(count) => count,
+                        Err(_) => {
+                            // If asset doesn't exist yet, create it with 0 count
+                            db.storage::<AssetsInfo>()
+                                .insert(&asset_id, &(*contract_id, **sub_id, 0))?;
+                            Some(Cow::Owned((*contract_id, **sub_id, 0)))
+                        }
+                    }
+                    .map(|info| {
+                        info.2
+                            .checked_add(*val)
```

Comment:
```suggestion
                        .map(|info| {
                            let (_, _, count) = *info;
                            count
                                .checked_add(*val)
```

May also affect the `Receipt::Burn` branch.

Reply:
Addressed in https://github.com/FuelLabs/fuel-core/pull/2445/commits/319d32d3deeaf32a2ef3ba886dc5decca577532e